BKTCLT-47 [impr] support createBucket to specific session ID

jonathan-gramain opened the pull request:
- Add optional query params to the createBucket HTTP request, in particular to allow creating the bucket on a specified RAFT session

- Minor unit test cleanup

- Allow creating buckets on RAFT session 0 via the Go client (sanity checks will be done by bucketd, only in case RS 0 is the map)

jonathan-gramain committed:
BKTCLT-47 [impr] support createBucket to specific session ID

Add optional query params to the createBucket HTTP request, in
particular to allow creating the bucket on a specified RAFT session.

Minor unit test cleanup.

jonathan-gramain committed:
BKTCLT-47 bump version

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/bucketclient/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

jonathan-gramain committed:
BKTCLT-47 [bf] Go client: allow CreateBucket with ?raftsession=0 query

Some deployment types do not have a map RAFT session, hence support
creating buckets on RS 0. The API now allows it.

jonathan-gramain commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.2`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.10`
* `development/7.4`
* `development/8.1`



Please check the status of the associated issue BKTCLT-47.

Goodbye jonathan-gramain.




*The following options are set:* **approve**
